### PR TITLE
Assignment7+8 @Disciple4GIT

### PR DESCRIPTION
**Pull request by nwnel-017**

### CODE REVIEW

**Comment by TechGlue** [comment]
I'll review this!

File: Assignment7+8/index.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    <br>
+    <hr>
+    <div class ="card">
+        <div class="card-title">A Joke<hr></div>
+        
+        <div class="card-content">
+            This is where my joke is going to go<br>
+        </div>   
+        <br>
+        <button class="another-joke">Another Joke</button>  <button class="joke-api">Joke API</button>
```
Comment (TechGlue) [inline Assignment7+8/index.html:43]: Remember that the Joke API button needs to be a <a> tag that is styled to look like a button.

**Review by TechGlue — COMMENTED** [review]
**General**

- The project should look like the image below ❌
    
    While it does kind of look like it the section for the menu bar is not at the top of the page like in the photo. As well as some of the button styling is off. 
    
- The project should have an index.html file ✔
    - With an index.js file in a js folder ❌
    - And a site.css file in a css folder (along with other css files) ✔
- There should be a `<head>`
    - With a favicon ❌
    - And a title ✔
- The `<body>` should use the middle 70% of the page ✔

### **Menu Bar**

- The page should have a title/menu bar ✔
    - Use Font Awesome for the hamburger icon ([https://fontawesome.com/start[](https://v2.jokeapi.dev/joke/Programming)](https://fontawesome.com/start)) ✔
    - When clicked, the menu should appear under it ✔
    - The menu should have four buttons with creative names (the buttons don't need to do anything) ✔
    
    Awesome job on the hamburger menu. 
    

### **Create a card with `<div>`s and CSS**

- Create a card from `<div>`s ✔
    - The card should have a header, body, and footer ✔
    - The CSS for this should be in the CSS file and work easily for another card. ✔

### **Show jokes on the card**

- Use the joke API from the first assignment to display a joke on a card ([https://v2.jokeapi.dev/joke/Programming[](https://cdn.jsdelivr.net/npm/axios/dist/axios.min.js)](https://v2.jokeapi.dev/joke/Programming)) ❌
    - Use Axios from the CDN for the API call. ([https://cdn.jsdelivr.net/npm/axios/dist/axios.min.js](https://cdn.jsdelivr.net/npm/axios/dist/axios.min.js))
    - If the joke API returns an error display a message to try again in a few moments ❌
    - The punch line should not show immediately, but appear after 4 seconds ❌
    - There should be a button to get another joke ✔ - button is present but not currently functioning.
    - There should be `<a>` tag that links to the joke api documentation page ❌
        - This `<a>` should be manually styled as a button ❌
- Buttons should have a background that is a gradient ❌
- All buttons (even the `<a>` that you made into a button) should have a hover effect ❌

## **Extra Credit**

Do the following:

1. Make the buttons in the menu do something creative ❌
2. Rather than just making the menu appear, have it fade in and slide down. ✔
3. Have the page jump to 100% width on a smaller screen and 50% on a larger screen with 70% on a regular screen.❌

Fundamentals
Ensure that:
This is built with raw HTML, CSS, and JavaScript, no external libraries aside from Axios and a font library ✔
The index.html file should work when opened from a file explorer ✔
Keep all files in appropriate folders ✔
Have warning free HTML, CSS, and JavaScript ❌
Use as little css as possible ✔
No inline CSS ❌
Validate HTML with: https://validator.w3.org/nu/#textarea (or similar) ❌
Validate CSS with: https://jigsaw.w3.org/css-validator/#validate_by_input (or similar) ✔
No errors when loading the page ✔
Choose simplicity over complexity ✔

File: LectureWeb/css/site.css [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    padding: 5px 10px;
+}
+
+table {
+    border-spacing: 0;
+    width: 100%;
+}
+
+.name{
+    width: 20%;;
```
Comment (TechGlue) [inline LectureWeb/css/site.css:55]: Double semicolons.

File: Assignment7+8/css/site.css [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    height:50px;
+    width:100px;
+    background-color: lightblue;
+  }
+
+  .joke-api
+  {
+    height:50px;
+    width:100px;
+    padding:10px
```
Comment (TechGlue) [inline Assignment7+8/css/site.css:79]: While I don't think the missing semicolon would matter for getting the proper results. It'd be best to include it to keep consistency.

File: Assignment7+8/index.html [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                            list.classList.toggle('newlist');     
+                        });
+                    </script>
+            </p>
+    </div>
+    <div style = "height:50px">
+        <p style = "font-size:40px">
+            <span style = "background-color: white; color:black;">The Joke Page</span>
+        </p>
+    </div>
```
Comment (TechGlue) [inline Assignment7+8/index.html:33]: Move the inline styles to your site.css file

File: Assignment7+8/index.html [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                        <button class="another-joke">Give Nate 100% on this assignment</button>
+                    </div>
+                    <script>
+                        let click = document.querySelector('.fa-clickable');
+                        let list = document.querySelector('.list');          
+                        click.addEventListener("click",()=>{
+                            list.classList.toggle('newlist');     
+                        });
+                    </script>
+            </p>
```
Comment (TechGlue) [inline Assignment7+8/index.html:27]: This tag is throwing an error. Removing it gets rid of the error but can't see the closing tag for the p. It looks correct but the HTML checker is throwing an error. I'd maybe experiment with changing the p tag to a div or double-checking your tags.

File: Assignment7+8/index.html [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+<html lang="en-us">
+<head>
+    <title>Joke Page</title>
+    <link rel="stylesheet" href="css/site.css">
+    <script src="https://kit.fontawesome.com/04f7016002.js" crossorigin="anonymous"></script>
+</head>
+
+<body>
+    <div class="page-title">
+            <p style = "font-size:40px">
```
Comment (TechGlue) [inline Assignment7+8/index.html:11]: Inline style

File: Assignment7+8/index.html [hunk 7]
```diff
@@ -0,0 +1,61 @@
+<!DOCTYPE html>
+<html lang="en-us">
+<head>
+    <title>Joke Page</title>
+    <link rel="stylesheet" href="css/site.css">
+    <link rel="icon" type="image/x-icon" href="favicon.ico">
```
Comment (mmwoodfo) [inline Assignment7+8/index.html:6]:
```suggestion
    <link rel="icon" type="image/x-icon" href="images/favicon.ico">
```
Put your icon in an images folder

File: Assignment7+8/index.html [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+                <a class="menu">Menu <i class="fa-solid fa-bars fa-clickable"></i></a>
+                    <div class="list dropdown" >
+                        <button class="dropdown-card">
+                            <a href="https://en.wikipedia.org/wiki/PS5">
+                                Best Console, hands down</a></button><br>
+                        <button class="dropdown-card">
+                            <a href="https://www.youtube.com/watch?v=dQw4w9WgXcQ&ab_channel=RickAstley">Stream Princess Bride</a></button><br>
+                        <button class="dropdown-card">
+                            <a href="http://notpron.com/">Hardest Internet Puzzle</a></button><br>
+                        <button class="dropdown-card" onclick ="alert('Did you not read the button?')">
```
Comment (mmwoodfo) [inline Assignment7+8/index.html:23]: Something to note, these links are what need to be clicked on. When clicking on the button, it doesn't do anything. This is generally a bad user experience. You should make sure that clicking anywhere on the button navigates you to the link by either styling an <a> tag to look like a button or using a click event with javascript.

**Pull request closed without merging**